implementations/rust/ockam/ockam_command/src/state.rs

```
@@ -0,0 +1,535 @@
+        identity: IdentityState,
+        name: &str,
+    ) -> anyhow::Result<NodeState> {
+        let path = {
+            let mut path = self.dir.clone();
+            path.push(name);
+            if !path.exists() {
+                return Err(anyhow::anyhow!("node `{name}` does not exist"));
+            }
+            path
+        };
+        Ok(NodeState::new(path, vault, identity))
+    }
+
+    fn vault_name(&self, name: &str) -> anyhow::Result<String> {
+        let mut path = self.dir.clone();
+        path.push(name);
+        path.push("vault");
+        let path = std::fs::canonicalize(&path)?;
+        file_stem(&path)
+    }
+
+    fn identity_name(&self, name: &str) -> anyhow::Result<String> {
+        let mut path = self.dir.clone();
+        path.push(name);
+        path.push("identity");
+        let path = std::fs::canonicalize(&path)?;
+        file_stem(&path)
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct NodeState {
+    version: String,
+    name: String,
+    path: PathBuf,
+    vault: VaultState,
+    identity: IdentityState,
+    // authorities: AuthoritiesConfig,
+    // setup: NodeSetupConfig, // a mix of the current commands.json with some additional fields to define services
+}
+
+impl NodeState {
+    fn new(path: PathBuf, vault: VaultState, identity: IdentityState) -> Self {
+        Self {
+            version: "2.0.0".to_string(),
+            name: path.file_name().unwrap().to_str().unwrap().to_string(),
+            path,
+            vault,
+            identity,
+        }
+    }
+
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn socket(&self) -> PathBuf {
+        self.path.join("socket")
+    }
+
+    pub fn stdout_log(&self) -> PathBuf {
+        self.path.join("stdout.log")
+    }
+
+    pub fn stderr_log(&self) -> PathBuf {
+        self.path.join("stderr.log")
+    }
+}
+
+#[derive(Default, Debug, Clone, Eq, PartialEq)]
+pub struct NodeConfig {
+    vault: Option<String>,
```

would call these `default_vault`  and `default_identity`

---

In my opinion, the `default` concept is only valid from the `NodeManager` perspective, where you can override the usage of the vault/identity found in the node's state directory with the ones from the command payload. 

At this layer though, given that there is only one possible vault and one possible identity, introducing the `default` notion might be misleading as there is a non-default.

---

disagree :shrug: .     Neither the vault or identity is **the** node' vault and identity.    They are the defaults that will be used if other is not specified